### PR TITLE
az-storage-blob: add page

### PR DESCRIPTION
**Pull request by WhileLoop**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/az-storage-blob.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# az storage blob
+
+> Manage blob storage containers and objects in Azure.
+> Part of `azure-cli`.
+> More information: <https://docs.microsoft.com/cli/azure/storage/blob>.
+
+- Download a blob to a file path:
+
+`az storage blob download --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} -f {{./local_name}}`
```
Comment (blueskyson) [inline pages/common/az-storage-blob.md:9]:
```suggestion
`az storage blob download --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} -f {{path/to/local_file}}`
```

File: pages/common/az-storage-blob.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+- Download blobs from a blob container recursively:
+
+`az storage blob download-batch --account-name {{storage_account_name}} --account-key {{storage_account_key}} -s {{container_name}} -d {{ path/to/remote}} --pattern {{filename_regex}}`
```
Comment (blueskyson) [inline pages/common/az-storage-blob.md:13]: Will the blobs be downloaded to the working directory?
```suggestion
`az storage blob download-batch --account-name {{storage_account_name}} --account-key {{storage_account_key}} -s {{container_name}} -d {{path/to/remote}} --pattern {{filename_regex}}`
```

File: pages/common/az-storage-blob.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+- Upload a local file to blob storage:
+
+`az storage blob upload --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} -f {{./local_name}}`
```
Comment (blueskyson) [inline pages/common/az-storage-blob.md:17]:
```suggestion
`az storage blob upload --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} -f {{path/to/local_file}}`
```

File: pages/common/az-storage-blob.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+- Delete a blob object:
+
+`az storage blob delete --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}}`
+
+- Generate a shared access signature for a blob:
+
+`az storage blob generate-sas --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} --permissions r --expiry {{Y-m-d'T'H:M'Z'}} --https-only`
```
Comment (blueskyson) [inline pages/common/az-storage-blob.md:25]: Is `r` a typo?
```suggestion
`az storage blob generate-sas --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} --permissions --expiry {{Y-m-d'T'H:M'Z'}} --https-only`
```

**Review by blueskyson — COMMENTED** [review]
Thanks for your contribution. Here are some suggestions.

File: pages/common/az-storage-blob.md (hunk 2 above)
Comment (WhileLoop) [reply pages/common/az-storage-blob.md]: Added destination flag.

File: pages/common/az-storage-blob.md (hunk 4 above)
Comment (WhileLoop) [reply pages/common/az-storage-blob.md]: r is the permission literal for read. I changed it a param.

**Comment by WhileLoop** [comment]
@blueskyson Updated per your suggestions.

File: pages/common/az-storage-blob.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+> Part of `azure-cli`.
+> More information: <https://docs.microsoft.com/cli/azure/storage/blob>.
+
+- Download a blob to a file path:
+
+`az storage blob download --account-name {{storage_account_name}} --account-key {{storage_account_key}} -c {{container_name}} -n {{path/to/blob}} -f {{path/to/local_file}}`
+
+- Download blobs from a blob container recursively:
+
+`az storage blob download-batch --account-name {{storage_account_name}} --account-key {{storage_account_key}} -s {{container_name}} -d {{ path/to/remote}} --pattern {{filename_regex}} --destination {{path/to/destination}}`
```
Comment (marchersimon) [inline pages/common/az-storage-blob.md:13]:
```suggestion
`az storage blob download-batch --account-name {{storage_account_name}} --account-key {{storage_account_key}} -s {{container_name}} -d {{path/to/remote}} --pattern {{filename_regex}} --destination {{path/to/destination}}`
```

**Comment by WhileLoop** [comment]
@marchersimon Suggestions applied.

File: pages/common/az-storage-blob.md (hunk 4 above)
Comment (blueskyson) [reply pages/common/az-storage-blob.md]: You can also use something like `{{r}}` to represent a commonly used permission set.

**Review by blueskyson — APPROVED** [review]

**Review by marchersimon — APPROVED** [review]
Thanks @WhileLoop

**Pull request merged**